### DISCUSSION TITLE
MATH heavy AUDIT

### DISCUSSION THREAD

**Discussion by sagoto80** [Cyfrin/security-and-auditing-full-course-s23#112] [Q&A] [answered]
is zksync era is full of math heavy bugs or business logic???

**Comment by cromewar** [comment]
Hello @sagoto80 can you reformulate please, take a look [here](https://github.com/Cyfrin/security-and-auditing-full-course-s23/blob/main/how-to-ask-a-question.md).

**Comment by akhilmanga** [comment, marked as answer]
Hi @sagoto80 , Yes it is based on math logic. You can start learning cryptography. so that it will be easy do zksync era audits.

Resources down below to know about zk:

Intro to zk (part-1) : https://www.zellic.io/blog/intro-to-zk-part-1/
Intro to zk (part-2): https://www.zellic.io/blog/intro-to-zk-part-2/
